src/synchronous_product/include/synchronous_product/search.h

```diff
@@ -133,10 +136,18 @@ class TreeSearch
 		SPDLOG_TRACE("Processing {}", *current);
 		if (is_bad_node(current)) {
 			current->state = NodeState::BAD;
+			if (incremental_labeling_) {
+				current->label = NodeLabel::BOTTOM;
+				current->label_propagate(controller_actions_, environment_actions_);
+			}
 			return true;
 		}
 		if (is_monotonically_dominated_by_ancestor(current)) {
 			current->state = NodeState::GOOD;
+			if (incremental_labeling_) {
+				current->label = NodeLabel::TOP;
+				current->label_propagate(controller_actions_, environment_actions_);
```

Comment:
This can be implicitly be tested by comparing results from incremental labeling search and standard search, if we want. Otherwise we can also set up an explicit test.